File: go-clean-architecture/assets/go-layered-service-template/internal/application/usecase/media/video/upload_video.go
package video

import (
	"context"

	"example.com/service/internal/application/service/shared"
	domainvideo "example.com/service/internal/domain/entity/media/video"
	repovideo "example.com/service/internal/domain/repository/media/video"
)

// UploadVideo is the use case responsible for registering a new video.
type UploadVideo struct {
	repo  repovideo.Repository
	idGen shared.IDGenerator
}

// NewUploadVideo builds the UploadVideo use case with its dependencies.
func NewUploadVideo(repo repovideo.Repository, idGen shared.IDGenerator) UploadVideo {
	return UploadVideo{
		repo:  repo,
		idGen: idGen,
	}
}

// UploadVideoInput defines the input for the UploadVideo use case.
type UploadVideoInput struct {
	Title    string
	FilePath string
}

// UploadVideoOutput defines the output for the UploadVideo use case.
type UploadVideoOutput struct {
	ID string
}

// Execute runs the UploadVideo use case.
func (uc UploadVideo) Execute(ctx context.Context, in UploadVideoInput) (UploadVideoOutput, error) {
	id := uc.idGen.NewID()

	videoEntity, err := domainvideo.New(id, in.Title, in.FilePath)
	if err != nil {
		return UploadVideoOutput{}, err
	}

	if err := uc.repo.Create(ctx, videoEntity); err != nil {
		return UploadVideoOutput{}, err
	}

	return UploadVideoOutput{ID: id}, nil
}

File: nodejs-clean-architecture/assets/ts-layered-service-template/src/application/usecase/media/video/upload_video/upload_video.ts
import { Video } from '#/domain/entity/media/video/video'
import type { VideoRepository } from '#/domain/repository/media/video/video.repository.interface'

import type { IdGenerator } from '#/application/service/shared/id_generator.interface'
import type {
  UploadVideoInput,
  UploadVideoOutput,
} from '#/application/usecase/media/video/upload_video/upload_video.type'

export class UploadVideo {
  constructor(
    private readonly repository: VideoRepository,
    private readonly idGenerator: IdGenerator,
  ) {}

  async execute(input: UploadVideoInput): Promise<UploadVideoOutput> {
    const video = new Video({ id: this.idGenerator.newID(), title: input.title })
    video.validate()

    await this.repository.save(video)

    return { id: video.id }
  }
}
